### PR TITLE
Add the ability to update properties on printers

### PR DESCRIPTION
**Pull request by tas50**
This follows the same pattern used in windows_printer_port

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/chef/resource/windows_printer.rb [hunk 1]
```diff
@@ -116,62 +116,62 @@ class WindowsPrinter < Chef::Resource
         introduced: "17.3",
         default_description: "The resource block name or the ipv4_address prepended with IP_."
 
+      def ps_bool_to_ruby(bool)
+        bool.to_s.downcase == "true" ? true : false
```
Comment (lamont-granquist) [inline lib/chef/resource/windows_printer.rb:120]: you don't need the ternary here at all.

**Review by lamont-granquist — APPROVED** [review]

File: lib/chef/resource/windows_printer.rb (hunk 1 above)
Comment (tas50) [reply lib/chef/resource/windows_printer.rb:120]: Then a "false" would end up being a nil in the resource which would blow up the type validation

Comment (lamont-granquist) [reply lib/chef/resource/windows_printer.rb:120]: `bool.to_s.downcase == "true"` is only going to be true or false

**Comment by tas50** [comment]
This didn't work and I'm failing to remember why now

**Comment by tas50** [comment]
Perhaps Neha could take this one at some point, but for now it's not working properly and I won't be finishing this one up.

**Pull request closed without merging**